foundation/eclipselink.core.test/src/it/java/org/eclipse/persistence/testing/tests/unitofwork/UnregisterUnitOfWorkTest.java

```diff
@@ -76,6 +81,10 @@ protected void test() {
             throw new TestErrorException("Deep unregister object did not work");
         }
 
+        if (!uow.getPrimaryKeyToNewObjects().values().stream().allMatch(Collection::isEmpty)) {
```

Comment:
After change at 5077 `!uow.getPrimaryKeyToNewObjects().isEmpty()` should be enough.